bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/menuitem/listener/LanguageItemSelectionListener.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright 2021-2022 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+
+package com.espressif.idf.ui.menuitem.listener;
+
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.widgets.MenuItem;
+import org.eclipse.ui.PlatformUI;
+
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.util.EclipseIniUtil;
+
+public class LanguageItemSelectionListener extends SelectionAdapter
+{
+	private static final String LANGUAGE_SWITCH = "-nl";
+	
+	EclipseIniUtil eclipseIniUtil;
+
+	public LanguageItemSelectionListener()
+	{
+		try
+		{
+			eclipseIniUtil = new EclipseIniUtil();
+		}
+		catch (Exception e)
+		{
+			IDFCorePlugin.getPlugin().getLog().error(e.getMessage());
+		}
+	}
+
+	@Override
+	public void widgetSelected(SelectionEvent selectionEvent)
+	{
+		MenuItem currentItem = ((MenuItem) selectionEvent.widget);
+		String code = currentItem.getText().split("-")[1].replaceAll("^[ \t]+|[ \t]+$", "");
+		if (eclipseIniUtil != null)
+		{
+			try
+			{
+				if (currentItem.getText().toLowerCase().contains("default"))
+				{
+					if (eclipseIniUtil.containsEclipseSwitchInEclipseIni(LANGUAGE_SWITCH))
+					{
+						eclipseIniUtil.removeEclipseSwitch(LANGUAGE_SWITCH);
+					}
```

Comment:
if it's already there, why do we want to remove and add it again. Looks like `removeEclipseSwitch` is also writing to a file system, this is also an overhead operation. See how we can enhance further?

Reply:
One more thing - if the eclipse is already in English, and the user selected the same language again, we can avoid restarting instead we can show some message. Ideally, it would be better if we don't show the default language in the menu items itself